Develop2

Angelina2412 opened the pull request:
Второе задание выполнила полностью, отчет прикрепила

Angelina2412 committed:
Add the created tests to develop_2

Angelina2412 committed:
add allure report

nikolay-977 commented on src/test/java/GetTheUserOrdersTest.java:
```diff
@@ -0,0 +1,68 @@
+//import dto.ForOrder;
```
Удали комментированный код

nikolay-977 commented on src/main/java/dto/UserWithoutEmail.java:
```diff
@@ -0,0 +1,29 @@
+package dto;
+
+public class UserWithoutEmail {
```
Не следует создавать отдельные классы UserWithoutEmail, UserWithoutName. Можно описать один класс User и при инициализации объекта не заполнять email либо namе

nikolay-977 commented on src/test/java/MakeTheUserTest.java:
```diff
@@ -0,0 +1,62 @@
+    public void makeTheUniqueUser(){
+        RandomUser randomUser = RandomUser.getRandomUniqUser();
+        Response message = apiClient.post("/api/auth/register",randomUser );
+        message.then().assertThat().body("success", Matchers.equalTo(true))
```
Исправь порядок проверки - сначала проверяем statusCode, затем остальное

nikolay-977 commented on src/test/java/MakeTheUserTest.java:
```diff
@@ -0,0 +1,62 @@
+import org.junit.After;
+import org.junit.Test;
+
+public class MakeTheUser {
```
Добавь удаление тестовых данных после выполнеия тестов. Иначе это может аффектить другие тесты. Используй для этого дополнительный метод с аннотацией @After.

nikolay-977 commented on src/test/java/LoginTheUserTest.java:
```diff
@@ -0,0 +1,58 @@
+
+    @DisplayName("Check status code after login")
+    @Test
+    public void loginWithSuccess() {
```
Используй инкапсуляцию. В src.main.java.api.client для каждой ручки создай классы CourierClient(для /api/v1/courier) и OrdersClient(для /api/v1/orders). Вынеси реализацию шагов в эти классы, а именно - в классах CourierClient и OrdersClient реализуй методы по примеру:

```
@Step("Get response for incorrect password") 
public Response getIncorrectPasswordRsponse(Object body) {
    return given()
    .header("Content-type", "application/json")
    .and()
    .body(body)
    .when()
    .post(loginCourier);
    .then();
}
```

И приведи кейсы к виду:

```
@Test 
@DisplayName("Check error message for incorrect password") 
public void testErrorMessageForIncorrectPassword(){ 
    CourierClient courierClient = new CourierClient();
    Response incorrectPasswordRsponse = courierClient.getIncorrectPasswordRsponse(new 
    AuthCourier(currentCourier.login,"*** WrongPassword ***"));
    incorrectPasswordRsponse.statusCode(404).and.assertThat().body("message", is("Учетная запись не найдена"));
}
```

Angelina2412 committed:
1) Удалила комментированный код
2) убрала лишнии классы в dto
3) исправила порядок проверки - сначала статус код, пото все остальное
4) использовала инкапусуляцию для MakeTheOrderTest, создала класс OrderClient, из-за ограниченности времени не успеваю реализовать для других классов, но смысл поняла

Angelina2412 committed:
1) Удалила комментированный код
2) убрала лишнии классы в dto
3) исправила порядок проверки - сначала статус код, пото все остальное
4) использовала инкапусуляцию для MakeTheOrderTest, создала класс OrderClient, из-за ограниченности времени не успеваю реализовать для других классов, но смысл поняла

Angelina2412 commented:
1) Удалила комментированный код
2) убрала лишнии классы в dto
3) исправила порядок проверки - сначала статус код, пото все остальное
4) использовала инкапусуляцию для MakeTheOrderTest, создала класс OrderClient, из-за ограниченности времени не успеваю реализовать для других классов, но смысл поняла
